### DISCUSSION TITLE
Cannot change the radio button value (ANTD)

### DISCUSSION THREAD

**Discussion by kaisiant** [react-hook-form/react-hook-form#2823] [Q&A] [answered]
Hello, the react hook form's magic does not help me change the radio button correctly but it gets the event :(
Should I use register instead of controller?

https://codesandbox.io/s/kind-glitter-vj9u1?fontsize=14&hidenavigation=1&theme=dark

**Comment by kaisiant** [comment, marked as answer]
Found the ans through this link https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r

<Controller
        control={control}
        name="AntdRadio"
        render={({ onChange, value }) => (
          <Radio.Group value={value} onChange={e => onChange(e.target.value)}>
            <Radio value={1}>A</Radio>
            <Radio value={2}>B</Radio>
            <Radio value={3}>C</Radio>
            <Radio value={4}>D</Radio>
          </Radio.Group>
        )}
      />

**Reply by PotatoEater0127** [reply]
An even cleaner solution would be inserting the `field` prop, instead of explicitly assigning `value` and `onChange`

``` javascript
<Controller
  control={control}
  name="AntdRadio"
  render={({ field }) => (
    <Radio.Group {...field}>
      <Radio>A</Radio>
      <Radio>B</Radio>
      <Radio>C</Radio>
   </Radio.Group>
  )}
/>
```
